product-backend/src/main/java/org/kainos/ea/controller/JobRoleController.java

```diff
@@ -0,0 +1,48 @@
+package org.kainos.ea.controller;
+
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.kainos.ea.db.JobRoleDao;
+import org.kainos.ea.exception.ErrorResponse;
+import org.kainos.ea.exception.FailedToCreateNewJobRoleException;
+import org.kainos.ea.model.JobRoleRequest;
+import org.kainos.ea.service.JobRoleService;
+import org.kainos.ea.service.JobRoleValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+
+@Tag(name = "Group A")
+@Path("/api")
+
+public class JobRoleController {
+    private final static Logger logger = LoggerFactory.getLogger(JobRoleService.class);
+    private final JobRoleService jobRoleService = new JobRoleService(new JobRoleDao(), new JobRoleValidator());
+
+    public JobRoleController() throws FailedToCreateNewJobRoleException {
+    }
+
+    @POST
+    @Path("/admin/job-roles")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createNewJobRole(JobRoleRequest jobRole) {
+        try {
+            return Response.ok(jobRoleService.createNewJobRole(jobRole)).build();
+        } catch (FailedToCreateNewJobRoleException e) {
+            logger.error("Failed to create new Job Role! Error: {}", e.getMessage());
+
+            return Response.serverError().entity(new ErrorResponse(e.getMessage())).build();
+        } catch (SQLException e) {
+            logger.error("Job Role already exists! Error : {}", e.getMessage());
```

Comment:
The exception is generic but the message is specific "Job Role already exists".

I you want to verify the job already exist you should reuse method from Adrian PR in JobRoleService
```
Optional<JobRole> jobRoleExist = jobRoleDao.getJobRoleById(jobRole.getId());
if (jobRoleExist.isPresent()) {
 throw new JobRoleAlreadyExistException();
}
```
Then in the controller catch it and return 400(with a message)